### PR TITLE
3783/remove arrow functions in methods

### PR DESCRIPTION
**Pull request by tatiana-scandi**
**Related issue(s):**
* Fixes https://github.com/scandipwa/scandipwa/issues/3783

**Problem:**
* Arrow functions can be overridden, which is a problem if ScandiPWA is used as base theme

**In this PR:**
* Added ESlint rule forbidding this
* Fixed around 200 cases when arrow functions were used.

### CODE REVIEW

File: packages/scandipwa/src/component/CategorySearch/CategorySearch.component.js [hunk 1]
```diff
@@ -1,68 +0,0 @@
-/**
- * ScandiPWA - Progressive Web App for Magento
- *
- * Copyright © Scandiweb, Inc. All rights reserved.
- * See LICENSE for license details.
- *
- * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
- * @package scandipwa/base-theme
- * @link https://github.com/scandipwa/base-theme
- */
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/CategorySearch/CategorySearch.component.js:10]: We don't use this component anywhere ?

**Review by AleksandrsKondratjevs — COMMENTED** [review]
This is very cool and useful :)

File: packages/scandipwa/src/component/CategorySearch/CategorySearch.component.js (hunk 1 above)
Comment (tatiana-scandi) [reply packages/scandipwa/src/component/CategorySearch/CategorySearch.component.js:10]: We don't :)
During previous refactoring related to prop-types I found 6 more not used components and also had to delete them.

Comment (AleksandrsKondratjevs) [reply packages/scandipwa/src/component/CategorySearch/CategorySearch.component.js:10]: Seems deprecated. Interesting if we have even more of those :face_exhaling:. Anyway cool that you deleted them.

**Comment by aleksandrakorolova** [comment]
1.Can't close video on PDP

**Steps:**
1. Open product with video - https://pr-3818.d25kd40decjpue.amplifyapp.com/default/travel-bag-1.html
2. Open video
3. Click anywhere outside the video 

**Actual result:**
Nothing happens on FE
Error message 'Popup.component.js:110 Uncaught TypeError: Cannot read properties of undefined (reading 'props')
    at value (Popup.component.js:110)
    at Proxy.value (ClickOutside.component.js:66)' appears in the console
https://drive.google.com/file/d/1YmDZsrYg7je7O57Exhz4TtRKvSTzBnp9/view?usp=sharing

**Expected result:**
Video is closed

**Notes:**
The same happens with all pop-up on website (write new review, add/edit address and etc)
__________________________
2.Can't share wishlist in my account

**Steps:**
1. Open Sign in form
2. Log in
3. Add product to Wishlist
4. Open Wishlist
5. Click on Share Wishlist
6. Enter valid email
7. Click on Share Wishlist

**Actual result:**
Nothing happens on the FE
Error message 'ShareWishlistForm.component.js:29 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'props')' appears in the console
Screenshot 2021-11-29 at 10 17 51

**Expected result:**
Whishlist is shared. User sees success message
__________________________
3.Show more button is not clickable on PLP

**Steps:**
1. Open PLP or search result page
2. Click on Show more in Layered navigation

**Actual result:**
Nothing happens on FE
Error message 'Uncaught TypeError: Cannot read properties of undefined (reading 'state')' appears in the console
Screenshot 2021-11-29 at 11 49 00

**Expected result:**
All options of attribute are displayed
Button 'Show more' changes to 'Show less'
Screenshot 2021-11-29 at 11 55 45
__________________________
4.Mobile: Filters button doesn't work

**Steps:**
1. Open PLP or search result page
2. Tap Filters

**Actual result:**
Nothing happens 
Error message 'Uncaught TypeError: Cannot read properties of undefined (reading 'props')' appears in the console
https://drive.google.com/file/d/1Pueln1vJ8audPywfuAp4EasOEcQxVlvG/view?usp=sharing

**Expected result:**
Filter pop-up opens
_____________________________
5.Mobile: Can't unwrap secttion About/Details/Reviews on PDP

**Steps:**
1. Open PDP
2. Tap About/Details/Review

**Actual result:**
Nothing happens 
Error message 'Uncaught TypeError: Cannot read properties of undefined (reading 'props')'appears in the console
https://drive.google.com/file/d/1Pu8J1hum16FPPffvBtIDKO1Vmj6KnZoJ/view?usp=sharing

**Expected result:**
Section is unwrapped

**Notes:**
The same problem with Links section for downloadable products and in my account 
__________________________
6.Mobile: Ooops page appears when opening NOT empty cart

**Steps:**
1. Open PDP
2. Add product to cart
3. Open Cart

**Actual result:**
Ooops page
Error message 'TypeError: Cannot read properties of undefined (reading 'containerFunctions')' appears in the console
Screenshot 2021-11-29 at 12 37 45

**Expected result:**
No errors

**Notes:**
The same situation with Checkout
__________________________
7.Mobile: Title is missing in my account pages

**Steps:**
1. Open Sign in page
2. Log in
3. Open My orders/downloadable/wishlist/newsletter subscription

**Actual result:**
Title is missing from my account pages
Screenshot 2021-11-29 at 13 14 15

**Expected result:**
Title is displayed on my account pages
__________________________
8.Mobile: Can't delete product from Wishlist via swiping

**Steps:**
1. Open Sign in page
2. Log in
3. Add product to Wishlist
4. Open Wishlist
5. Swipe product card

**Actual result:**
Nothing happens on FE
Error messages 'Uncaught TypeError: Cannot read properties of undefined (reading 'props')' and 'Uncaught TypeError: Cannot read properties of undefined (reading 'setAnimationSpeedStyle')' appear in the console
Screenshot 2021-11-29 at 13 24 48

**Expected result:**
Product is deleted via swiping 
No errors

**Comment by tatiana-scandi** [comment]
Fixed 1-6, 8
7 - didn't reproduce for me

**Comment by aleksandrakorolova** [comment]
1. Fixed
2. Fixed
3. Fixed
4. Fixed
5. Fixed
6. Fixed
7. Also, can't reproduce now
8. Fixed
9. Button 'Change password' doesn't work in my account
Screenshot 2021-12-03 at 10 46 37

10. User sees error 'Uncaught TypeError: Cannot read properties of undefined (reading 'props')' in the console and order is completed when opening T&C popup
https://drive.google.com/file/d/1xFqMramBZSchI-92tguKXsKOLnYz8Mma/view?usp=sharing

**Comment by tatiana-scandi** [comment]
Fixed issue 9 and 10 + resolved merge conflicts.

As for issue 10, error in console was related to this PR, but the one with form mistakenly submitted was due to "read more" button having type="submit". And when button in form has type submit (which is the default `<Button/>` type in ScandiPWA), click on it submits the whole form.

**Review by AleksandrsKondratjevs — APPROVED** [review]

**Comment by aleksandrakorolova** [comment]
9. It works only if user entered valid current and new password
Please, fix it if user entered invalid current password and/or new password

Error message '_Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'props')at value (MyAccountCustomerPopup.container.js:83)_' appears in the console
Screenshot 2021-12-07 at 12 13 16

**After master was merged:** user sees error message 'The account is locked.' when tried to change password with invalid confirm and new password
Screenshot 2021-12-07 at 14 14 05

10. Fixed

**Comment by carinadues** [comment]
@tatiana-scandi says the last issue in 9 is not related to this PR. @aleksandrakorolova can you move it to a separate task, please?

**Pull request merged**